File: teams/tl-haskell/two.hs
module BrisFunctional.Euler2
       ( euler2        
       , prop_euler2
       ) where

import Test.QuickCheck

prop_euler2 :: Property
prop_euler2 = property $ euler2 == 4613732

euler2 :: Integer
euler2 = sum [x | x <- takeWhile (<= 4000000) fibs, even x]

fibs :: [Integer]
fibs = 1 : 2 : zipWith (+) fibs (tail fibs)

File: teams/tl-js/src/two.js
var euler2 = function () {
    return fibs (1, 1, 0);
};

var fibs = function (a, b, acc) {
    if (a >= 4000000) return acc;
    else if (a % 2 === 0) return fibs(a + b, a, acc + a)
    else return fibs(a + b, a, acc);
};
